compiler/ir/backend.jvm/codegen/src/org/jetbrains/kotlin/backend/jvm/intrinsics/Not.kt

```diff
@@ -40,6 +39,13 @@ object Not : IntrinsicMethod() {
             markLineNumber(expression)
             value.discard()
         }
+
+        override fun materializeAt(target: Type, irTarget: IrType, castForReified: Boolean) {
+            value.materializeAt(Type.BOOLEAN_TYPE, codegen.context.irBuiltIns.booleanType, false)
+            mv.iconst(1)
+            mv.xor(Type.INT_TYPE)
+            StackValue.coerce(Type.BOOLEAN_TYPE, target, mv)
```

Comment:
the fact that you're not using `irTarget` here seems suspicious, because it could be an inline class type. wasn't there a 5-argument version of coerce?

Reply:
Is that possible? Can you please give me an example where `irTarget ` is an inline class type?